### PR TITLE
Scala 2.13.0-M4

### PR DESCRIPTION
**Pull request by xuwei-k**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -22,16 +30,35 @@ lazy val core = crossProject(JVMPlatform, JSPlatform).in(file(".")).
...
+    mimaPreviousArtifacts := {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((2, v)) if v >= 13 =>
+          Set()
+        case _ =>
+          mimaPreviousArtifacts.value
+      }
+    },
     mimaBinaryIssueFilters ++= Seq(
+      ProblemFilters.exclude[IncompatibleMethTypeProblem]("scodec.DecoderFunctions.decodeCollect"),
```
[the remark this answers is not in the document]
Comment (xuwei-k) [reply build.sbt]: Yes real binary incompatibilities :disappointed_relieved:

File: .travis.yml [hunk 2]
```diff
@@ -8,9 +8,9 @@ cache:
 language: scala
 
 scala:
-  - 2.10.6
```
Comment (xuwei-k) [inline .travis.yml]: We can't continue Scala 2.10 support if use `scala-collection-compat` https://github.com/scala/scala-collection-compat/issues/1

**Comment by xuwei-k** [comment]
binary compatible version. but some tricky hacks.
https://github.com/xuwei-k/scodec/commit/84e5f307bd1f36a2d0f3e4690a3331e8356a6091

**Comment by mpilquist** [comment]
Wow, awesome! Normally I'd prefer the backwards compatible version but in this case, I'm thinking we release scodec 1.11.0-M1 and drop 2.10 support. That way we don't need our own version of CanBuildFrom. Also, Shapeless is dropping 2.10 support soon as well. What do you think?

**Comment by milessabin** [comment]
FYI: shapeless support for 2.10 will continue for 2.3.x ... I won't be dropping 2.10 support until 2.4.0.

File: build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (smarter) [reply build.sbt]: scala-collection-compat 0.2.0 has been released which should solve the binary incompatibilities: https://github.com/scala/scala-collection-compat/pull/115

**Comment by alexarchambault** [comment]
Anything prevents this to be merged? And switching to `2.13.0-M5` at the same time.

**Comment by mpilquist** [comment]
Not really sure how to proceed here. I think we should maintain 2.13 support on a branch until the final release, so as to avoid introducing binary compatibility problems on the 2.11 or 2.12 builds as a result of changes in scala-collection-compat.

**Comment by mpilquist** [comment]
Merged as part of #138

**Pull request closed without merging**